ext/drbbdup/drbbdup.c

```diff
@@ -844,7 +859,11 @@ drbbdup_insert_dynamic_handling(void *drcontext, app_pc translation_pc, void *ta
             instrlist_meta_preinsert(bb, where, instr);
         }
     }
-    /* XXX i#4134: Insert code for dynamic handling here. */
+
+    if (opts.is_stat_enabled) {
+        /* Insert clean call so that we can lock stat_mutex. */
+        dr_insert_clean_call(drcontext, bb, where, drbbdup_inc_bail_count, false, 0);
```

Comment:
How about drx_insert_counter_update() with DRX_COUNTER_LOCK

Reply:
Only downside is that on 32-bit you are limited to uint causing potential overflow for long running CPU-intensive apps.

Reply:
Yeah this bit me too in the drwrap_stats (that stat was ok being small so I shrunk it for 32-bit as a workaround).  I filed #4215 on adding better atomics support.

Reply:
I added a fixme comment.